tests/phpunit/tests/autoloader-classmap.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ * Tests related to the WP_Autoload class.
+ *
+ * @group basic
+ */
+class Tests_Autoloader_Classmap extends WP_UnitTestCase {
+
+	/**
+	 * Test that all classes in the classmap are lowercase.
+	 *
+	 * @dataProvider data_autoloader_classmap_is_lowercase
+	 *
+	 * @param string $class_name Class name.
+	 */
+	public function test_autoloader_classmap_is_lowercase( $class_name ) {
+		$this->assertSame( strtolower( $class_name ), $class_name, "Class name '$class_name' is not lowercase." );
+	}
+
+	/**
+	 * Data provider for test_autoloader_classmap_is_lowercase.
+	 *
+	 * @return array Data provider.
+	 */
+	public function data_autoloader_classmap_is_lowercase() {
+		$class_names = array_keys( WP_Autoload::CLASSES_PATHS );
```

Comment:
This looks like it can use [WP_UnitTestCase_Base::text_array_to_dataprovider()](https://github.com/wordpress/wordpress-develop/blob/a2e69ce7136ccf0a40c2bdede80b77903e4d11e9/tests/phpunit/includes/abstract-testcase.php#L1149), which will also mean the datasets are named.

Reply:
Oooh nice! I was not aware of this method. Done 👍